src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -669,6 +681,31 @@ private static void ValidateReadArgs(byte[] buffer, int offset, int count)
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) { throw new NotSupportedException(); }
             public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) { throw new NotSupportedException(); }
         }
+
+
+        private static void WriteToStream(Stream stream, string content, Encoding encoding)
+        {
+            if (ReferenceEquals(encoding, Encoding.Latin1) ? content.Length <= 512
+                : (ReferenceEquals(encoding, Encoding.UTF8) && content.Length <= 128))
+            {
+                // Fast-path for the expected common cases
+                Span<byte> buffer = stackalloc byte[512];
```

Comment:
These limits are building in the encoding's max char-to-byte size, and it's relying on that for correctness?  That's not obvious from the magic numbers here.  It'd be good to put them in consts with appropriate comments.

Reply:
In the changes I'm making for SocketsHttpHandler (not this PR), I make the assumption that converting UTF16 code units (chars) to UTF8 bytes is guaranteed to be at most 3 bytes per code unit (to avoid calling into GetByteCount every time). @GrabYourPitchforks Does that assumption hold? (as in worst-case every char is a 3-byte replacement character)

The bounds (128 => 512) can be tightened further here in that case. I will extract these to consts.

Reply:
Changed to `GetMaxByteCount`